lib/FrameManager.js

```diff
@@ -291,55 +291,93 @@ class Frame {
   }
 
   /**
-   * @param {string} filePath
-   * @return {!Promise<*>}
+   * @param {Object} options
+   * @return {!Promise}
    */
-  async injectFile(filePath) {
-    let contents = await new Promise((resolve, reject) => {
-      fs.readFile(filePath, 'utf8', (err, data) => {
-        if (err) return reject(err);
-        resolve(data);
+  async addScriptTag(options) {
+    if (typeof options.url === 'string')
+      return this.evaluate(addScriptUrl, options.url);
+
+    if (typeof options.path === 'string') {
+      let contents = await new Promise((resolve, reject) => {
+        fs.readFile(options.path, 'utf8', (err, data) => {
+          if (err) return reject(err);
+          resolve(data);
+        });
       });
-    });
-    contents += `//# sourceURL=` + filePath.replace(/\n/g,'');
-    return this.evaluate(contents);
-  }
+      contents += `//# sourceURL=` + options.path.replace(/\n/g, '');
 
-  /**
-   * @param {string} url
-   */
-  async addScriptTag(url) {
-    return this.evaluate(addScriptTag, url);
+      return this.evaluate(addScriptContent, contents);
+    }
+
+    if (typeof options.content === 'string')
+      return this.evaluate(addScriptContent, options.content);
 
```

Comment:
Ditto: let's throw if neither of the options was given (would help with migration)